lib/auth/join/boundkeypair/boundkeypair.go

```diff
@@ -185,12 +201,9 @@ func (c *ClientState) SignerForPublicKey(authorizedKeysBytes []byte) (crypto.Sig
 
 	// Check the active key first, if available.
 	if c.PrivateKey != nil {
-		activePubKeyBytes, err := c.ToPublicKeyBytes()
-		if err != nil {
-			return nil, trace.Wrap(err)
-		}
+		activePubKey := c.PrivateKey.Public()
 
-		equal, err := pubKeyEqual(desiredPubKey, activePubKeyBytes)
+		equal, err := pubKeyEqual(desiredPubKey, activePubKey)
```

Comment:
Not sure how this bug slipped through, I guess the underlying `Equals()` impl must support `[]byte`? Gotta love comparing `any`s...